### PR TITLE
Added the ability to suppress exceptions when a reboot is required

### LINKED ISSUE #109: DismRebootRequiredException is thrown for non-exceptional cases
When enabling a feature (or trying to perform other operations) succeeds, but requires a reboot, a DismRebootRequiredException is thrown. Even if the consumer catches that exception, subsequent calls fail with the same exception. 

Would a change to this behavior be considered?

### PR DESCRIPTION
**Pull request by jhennessey**
Closes #109

### CODE REVIEW

**Review by jeffkl — COMMENTED** [review]
Code looks perfect, unit tests and everything!  Thanks for the contribution!

I'm trying to figure out why the tests seem to be hanging in CI, will approve as soon as I get it working or figure out what's up.

**Comment by jhennessey** [comment]
> I'm trying to figure out why the tests seem to be hanging in CI, will approve as soon as I get it working or figure out what's up.

Ah, it looks like the issue is due to the fact them I'm spinning up DISM sessions in my test and some of the other tests do as well. I will update my new test to inherit from `TestBase`.

**Comment by jeffkl** [comment]
Yeah looks like it, I almost have it working if you want me to push a commit, or you can work on it

**Comment by jhennessey** [comment]
I think I committed a fix right before your comment...All are passing now :)

File: src/Microsoft.Dism.Tests/DismSessionTest.cs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+// Licensed under the MIT license.
+
+using Shouldly;
+using System;
+using System.Collections.Generic;
+using Xunit;
+
+namespace Microsoft.Dism.Tests
+{
+    public class DismSessionTest : TestBase
```
Comment (jeffkl) [inline src/Microsoft.Dism.Tests/DismSessionTest.cs:12]: Can you derive from `DismTestBase` instead, it calls Initialize() and Shutdown for you and provides the log to the TestOutputHelper.  Then you can remove the Initialize() and Shutdown() in `SessionOptionsBehavior`

File: src/Microsoft.Dism.Tests/DismSessionTest.cs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            {
+                (_) => DismApi.OpenOnlineSession(),
+                (_) => DismApi.OpenOfflineSession(DismApi.DISM_ONLINE_IMAGE),
+                (_) => DismApi.OpenOfflineSession(DismApi.DISM_ONLINE_IMAGE, null, null),
+            };
+
+            private readonly List<Func<DismSessionOptions, DismSession>> _publicSessionExMethods = new List<Func<DismSessionOptions, DismSession>>
+            {
+                (options) => DismApi.OpenOnlineSessionEx(options),
+                (options) => DismApi.OpenOfflineSessionEx(DismApi.DISM_ONLINE_IMAGE, options),
```
Comment (jeffkl) [inline src/Microsoft.Dism.Tests/DismSessionTest.cs:87]: This is technically not opening a session to an offline image, I'm not sure if that matters.  You could modify the test to mount an image and then get the offline session, but again, not sure if that matters.  What do you think?

**Comment by jeffkl** [comment]
> I think I committed a fix right before your comment...All are passing now :)

Perfect, just 2 comments then

File: src/Microsoft.Dism.Tests/DismSessionTest.cs (hunk 2 above)
Comment (jhennessey) [reply src/Microsoft.Dism.Tests/DismSessionTest.cs:87]: Yeah, I sort of gamed the system there. My thought was to make it as quick as possible while still going through the public methods so I could test the new session options are applied (or aren't). So, this was intentional...does is make sense?

Comment (jeffkl) [reply src/Microsoft.Dism.Tests/DismSessionTest.cs:87]: Yeah, its still testing the code path so let's just leave it as is

**Review by jeffkl — APPROVED** [review]

**Pull request merged**